solidity/dashboard/src/components/BondingSection.jsx

```diff
@@ -76,81 +74,51 @@ export const BondingSection = ({ data }) => {
 
 export default React.memo(BondingSection)
 
-const AvailableEthCell = React.memo(({ availableETH }) => {
-  return (
-    <>
-      <span
-        className="text-big text-grey-70"
-        style={{
-          textAlign: "right",
-          padding: "0.25rem 1rem",
-          paddingLeft: "2rem",
-          borderRadius: "100px",
-          border: `1px solid ${colors.grey20}`,
-          backgroundColor: `${colors.grey10}`,
-        }}
-      >
-        {availableETH}
-      </span>
-      &nbsp;ETH
-    </>
-  )
-})
-
 const ActionCell = React.memo(
   ({ availableETH, availableETHInWei, operatorAddress, isWithdrawable }) => {
-    const { ModalComponent, openModal, closeModal } = useModal()
-
-    const web3Context = useWeb3Context()
-    const showMessage = useShowMessage()
-
-    const withdrawAllUnbondedAmount = useCallback(
-      async (onTransactionHashCallback) => {
-        try {
-          await tbtcAuthorizationService.withdrawAllEthForOperator(
-            web3Context,
-            { operatorAddress, availableETH },
-            onTransactionHashCallback
-          )
-          showMessage({
-            type: messageType.SUCCESS,
-            title: "Success",
-            content: "Withdrawal of ETH transaction has successfully completed",
-          })
-        } catch (error) {
-          showMessage({
-            type: messageType.ERROR,
-            title: "Withdrawal of ETH has failed ",
-            content: error.message,
-          })
-          throw error
-        }
-      },
-      [operatorAddress, availableETH, showMessage, web3Context]
-    )
+    const { openModal, closeModal, ModalComponent } = useModal()
+    const [action, setAction] = useState("withdraw")
+    const title = action === "add" ? "Add ETH" : "Withdraw ETH"
 
     return (
       <>
-        <ModalComponent title="Add ETH">
-          <AddEthModal
-            operatorAddress={operatorAddress}
-            closeModal={closeModal}
-          />
+        <ModalComponent title={title}>
+          {action === "add" ? (
+            <AddEthModal
+              operatorAddress={operatorAddress}
+              closeModal={closeModal}
+            />
+          ) : (
+            <WithdrawEthModal
+              operatorAddress={operatorAddress}
+              availableETH={availableETH}
+              closeModal={closeModal}
+            />
+          )}
         </ModalComponent>
         <div
           className="flex row center space-between"
           style={{ marginLeft: "auto" }}
         >
-          <Button onClick={openModal} className="btn btn-secondary btn-sm">
+          <Button
+            onClick={() => {
```

Comment:
Let's extract this fn and add the id prop to the button eg.

```
const onBtnClick = (event) => {
 setAction(event.currentTarget.id)
 openModal()
}

<Button
 id="add"
 onClick={onBtnClick}
 <other props here>
/>
```

Reply:
done